### ISSUE TITLE
New test added fails

### ISSUE THREAD

**Issue by diasdavid** [ipfs/js-ipfs-unixfs-engine#181]
@alanshaw I completely missed to see the status on CI :( (shame on me). Seems that the test you added detected a problem. Does it run for you?

https://circleci.com/gh/ipfs/js-ipfs-unixfs-engine/567

```
 1) IPFS UnixFS Engine builder allows multihash hash algorithm to be specified:
     Uncaught TypeError: Argument must be a Buffer or Uint8Array
      at Buffer.equals (buffer.js:635:11)
      at ipldResolver.get (test/test-builder.js:48:28)
      at _get (node_modules/ipld-resolver/src/index.js:123:9)
      at node_modules/async/internal/once.js:12:16
      at next (node_modules/async/waterfall.js:21:29)
      at node_modules/async/internal/onlyOnce.js:12:16
      at r.util.deserialize (node_modules/ipld-resolver/src/index.js:361:13)
      at multihashing (node_modules/ipld-dag-pb/src/dag-node/create.js:61:7)
      at Multihashing.Multihashing.digest (node_modules/multihashing-async/src/index.js:33:5)
      at setImmediate (node_modules/multihashing-async/src/utils.js:8:7)
      at Immediate._onImmediate (node_modules/async/internal/setImmediate.js:27:16)
```

**Comment by alanshaw** [comment]
That's my fault, I'll fix

**Comment by alanshaw** [comment]
@diasdavid fixed here #182
